Support Debian 13 for HT apache servers

rrotter opened the pull request:
~This is what I had to change to get puppet to apply sucessfully on Debian 12.~
Changes to support Debian 13.

rrotter committed:
HT Debian 13 support: php

- rm php-geoip: not available in Debian 12
- add php-raphf: php-http appears to functionally depend on php-raphf,
  but it's no longer a declared dependency in Debian 12
- HathiTrust catalog php 8: get pkgs even w/ Dockerfile
  See https://github.com/hathitrust/catalog/tree/ETT-636-trixie (which
  should work for debian 12 / bookworm as well)

- do not rm old php packages: we won't support upgrading php deps, only
  fresh installs

Co-authored-by: Aaron Elkiss <aelkiss@umich.edu>

aelkiss commented:
I'll have some more edits based on what I suspect to actually be in use.

aelkiss reviewed (approved):
I think this should be good to go on preview but we'll want to wait on rolling this out more widely until probably mid-January so that everybody has sufficient time to test.

rrotter committed:
HT Debian 13 support: update dependencies

- replace netpbm-sf w/ netpbm: newer version on bookworm and later
- use default-jre-headless: so we don't have to list the version for
  each os revision
- rm rdist from htvm role: no longer in debian

rrotter committed:
HT Debian 13 support: perl

- switch available perl modules from cpan to debian
- rm unused perl modules

Co-authored-by: Aaron Elkiss <aelkiss@umich.edu>

rrotter committed:
Debian 13 support: import ImageMagick-7 policy.xml

- use max of our current default and debian 13 defaults
- skip settings that are not set in debian 13 defaults
- place in new directory (breaks debian < 13)

aelkiss commented on manifests/profile/hathitrust/perl.pp:
```diff
@@ -139,18 +146,11 @@
     'File::Pairtree',
     'CGI::Application::Plugin::Routes',
     'Algorithm::LUHN',
-    'OAuth::Lite',
```
These weren't required by our Dockerfile and likely aren't needed any more at all:

* `OAuth::Lite` (@moseshll the CRMS Dockerfile depends on it but I don't see it in use anywhere in the code)
* `Sub::Uplevel`
* `Devel::Cycle`
* `Test::Memory::Cycle`
* `Mozilla::CA`

@moseshll Are you aware of any needs for these modules? (Some could be transitive dependencies of things we still need to install via CPAN?)

aelkiss replied:
Also available via Debian:

* `libip-geolocation-mmdb-perl`
* `libmarc-perl`
* `libmarc-xml-perl` (includes `MARC::File::XML`)

Apparently not used AFAICT:

* `CGI::Application::Plugin::Routes`

moseshll replied:
OAuth::Lite used to be used with Data API in CRMS but that code was removed. The dependency can be removed AFAICT. The rest were probably copypasta from `feed` or some other Perl repo.

aelkiss commented on manifests/profile/hathitrust/perl.pp:
```diff
@@ -70,6 +70,7 @@
     'libio-html-perl',
     'libio-socket-ssl-perl',
     'libio-string-perl',
+    'libip-geolocation-mmdb-perl',
```
only available on debian 13, not debian 12. Are we expecting this needs to work specifically on debian 12 or are we aiming to go straight to debian 13? Do we need to add any conditionality? Or should we just keep some things installed via CPAN until everything is on debian 13?

rrotter replied:
I want to be done w/ Debain 12.

aelkiss commented on manifests/profile/hathitrust/dependencies.pp:
```diff
@@ -20,7 +20,7 @@
       'kakadu',
```
we don't have kakadu packaged for debian 13, and I think it's time to cut bait and see if grokj2k will work for production purposes there.

rrotter replied:
I added grokj2k to the PR. So far it was just installed by hand on one test host.

aelkiss committed:
HT Debian 13 support: rm kakadu, add grokj2k

rrotter commented on manifests/profile/hathitrust/php.pp:
```diff
@@ -39,16 +47,16 @@
+      'HTTP_Session2'         => { ensure => 'absent', package_prefix => '', provider => 'pear' },
+      'Log'                   => { ensure => 'absent', package_prefix => '', provider => 'pear' },
+      'PHP_Compat'            => { ensure => 'absent', package_prefix => '', provider => 'pear' },
+      'Structures_LinkedList' => { ensure => 'absent', package_prefix => '', provider => 'pear' },
```
These are only installed if we installed them, right (i.e. not part of Debian)? Okay if we remove the `ensure => 'absent'` lines, since we'll be imaging new hosts w/ Debian 13?
```suggestion
```

aelkiss replied:
That should be correct.

rrotter committed:
HT: imgsrv sudoers requires sudo package

first puppet run fails w/o this

rrotter commented:
Rebased and squashed.
